test/dns_test.erl

```diff
@@ -32,6 +32,65 @@ message_other_test() ->
     Bin = dns:encode_message(Msg),
     ?assertEqual(Msg, dns:decode_message(Bin)).
 
+message_txt_test() ->
+    QName = <<"lorem.example.org">>,
+    Lorem = ten_lorem_ipsums_in_chunks(),
+    Qs = [#dns_query{name = QName, type = ?DNS_TYPE_TXT}],
+    As = [
+        #dns_rr{
+            name = QName,
+            type = ?DNS_TYPE_TXT,
+            ttl = 0,
+            data = #dns_rrdata_txt{txt = Lorem}
+        }
+    ],
+    QLen = length(Qs),
+    ALen = length(As),
+    Msg = #dns_message{qc = QLen, anc = ALen, questions = Qs, answers = As},
+    Bin = dns:encode_message(Msg),
+    ?assertEqual(Msg, dns:decode_message(Bin)).
+
+truncated_txt_test() ->
+    QName = <<"lorem.example.org">>,
+    Lorem = ten_lorem_ipsums_in_chunks(),
+    Qs = [#dns_query{name = QName, type = ?DNS_TYPE_TXT}],
+    As = [
+        #dns_rr{
+            name = QName,
+            type = ?DNS_TYPE_TXT,
+            ttl = 0,
+            data = #dns_rrdata_txt{txt = Lorem}
```

Comment:
Is it me, or you are already assigning a 255-chunked "Lorem" here? If so, you won't be able to verify that the encoding does the splitting, as the input is already split.

Unless I interpreted the `ten_lorem_ipsums_in_chunks()` incorrectly.

Reply:
Added a test for a contiguous case, `dns:encode_message/1` just crashes there :)

Reply:
> Added a test for a contiguous case, `dns:encode_message/1` just crashes there :)

That's the important bit we discussed yesterday: I assume what should happen here is a that a long text is split (formatted) into 255-max chunks.

Reply:
Where else is the splitting happening otherwise?